### ISSUE TITLE
Make display language configurable in the options

### ISSUE THREAD

**Issue by SaltfishAmi** [birchill/10ten-ja-reader#701]
I recall discussing about this topic somewhere sometime but anyway I think this needs to be a separate issue.

My OS and browser is in English because it's their original language and (luckily) I can understand it.
But I learned Japanese with Chinese and Japanese itself and therefore I can't understand the grammar terms written in English.
In the past few months I built my own variation of Rikaichamp and removed the English localization file (along with a few other customizations) to make it display every grammar term in Japanese. Not an elegant solution is it?

I don't want to switch my browser to Japanese either because oh god, it's a hell of katakana gairaigo in every piece of modern software in Japanese! I don't want to torture myself.

It would be really great to have language configuration in the options.

**Comment by birtles** [comment]
Yeah, it would be really nice to do this. (Personally I normally use the Japanese version of Firefox and every time I go to take screenshots I need to find an English version somewhere first so this would help me a lot too!)

Unfortunately the WebExtension [`i18n.getMessage()`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/i18n/getMessage) API doesn't allow specifying the locale and I haven't been able to find a way fetching other locale's strings  short of writing an independent implementation of the `i18n` API 😓

**Comment by birtles** [comment]
I suppose if we narrowed the scope to just the content of the popup that would be a bit easier but we'd still need to somehow get all those strings into the content script without bloating it too much.

**Comment by shirakaba** [comment]
Given that users may have an OS in one language, a browser in another, and may wish to go in and out of full immersion at any moment, what if we made the language explicitly configurable in the settings (and, when configured, use some generic non-internationalised message-passing API), falling back to `i18n.getMessage()` only when the language hadn't been explicitly configured?

**Comment by birtles** [comment]
So I _think_ that content scripts can read extension URLs (we _might_ need to add them to `web_accessible_resources` but I don't think so). So hopefully we could just look up the locale files directly from the content script.

If not our options for getting the strings into the content script might be one of the following:

1. Pass them along with config settings
   * Have webpack load the required JSON files on-demand using dynamic imports from an extension URL from the background script
   * Make sure all the strings used by the content script are prefixed with `content_`
   * Pass the set of content strings to the content script along with the other config options after stripping out unnecessary metadata like the `description` field
1. Pre-process the JSON files as part of the build process and bundle all the required strings into the content script itself. Given that we only support three languages that might be tractable.

Once we did that we'd still need to implement compatible placeholder substitution. Hopefully that wouldn't swell the content script by too much though.

For the background script and options page we could use a similar approach.

The tricky part about loading data from an extension URL is that it's async but `i18n.getMessage()` is sync. We don't want to rewrite all our call sites for `i18n.getMessage()` to be async so I guess we'd probably want to make a little `i18n.ts` module that:

* Exposes a `getMessage()` API identical to `i18n.getMessage()`
* Exposes a `setUILanguage()` API that triggers a load of the locale data and sets some internal flags to indicate if we are loading or not
* Falls back to calling `i18n.getMessage()` if we are still loading locale data (or if no language has been set)
